src/toil/cwl/cwltoil.py

```diff
@@ -814,6 +882,18 @@ def writeGlobalFileWrapper(file_store: AbstractFileStore, fileuri: str) -> str:
     return file_store.writeGlobalFile(schema_salad.ref_resolver.uri_file_path(fileuri))
 
 
+def remove_empty_listings(rec: CWLObjectType) -> None:
+    if rec.get("class") != "Directory":
+        finddirs = []  # type: List[CWLObjectType]
+        visit_class(rec, ("Directory",), finddirs.append)
+        for f in finddirs:
+            remove_empty_listings(f)
```

Comment:
Interesting. Is it `cwltool` code or `cwltoil` code that judges a `class: Directory` based upon if there is a `listing` or not; and can't handle an empty `listing`?

Reply:
I don't think it's necessarily a problem on the cwltool end of things.